reprounzip-singularity/Singularity_unpacker.py

```diff
@@ -0,0 +1,236 @@
+import copy
+import tarfile
+import os
+import sys
+import yaml
+import subprocess
+
+# Open outer tar, the RPZ file
+SINGULARITY_DIR = "../../.singularity.d"
+RUN_ENV_FILE="90-environment.sh"
+ENV_DIR = SINGULARITY_DIR+"/env"
+OVERLAY_IMAGE = "repro_overlay.img"
+IMAGE_TAR_FILE = "new.tar.gz"
+MOUNT_DIR = "/mnt/"
+source_config="METADATA/config.yml"
+APP_DIR = "../../app_structure/sample_app"
+SCIF_DIR = "../../app_structure/scif"
+APP_ENV_DIR = APP_DIR+"/scif/env"
+
+
+def write_env_file(env,env_file,apps=False):
+    if apps:
+        env_file = os.path.join(APP_ENV_DIR, env_file)
+    else:
+        env_file = os.path.join(ENV_DIR, env_file)
+    with open(env_file, 'w+') as f:
+        for key,value in env.items():
+            f.write(key+"='"+value+"'\n")
```

Comment:
You are going to need more robust quoting for values (and possibly keys). Simply enclosing in ' will break if the value contains ' (and backslashes, ", $, etc).

You can use https://github.com/ViDA-NYU/reprozip/blob/1.0.13/reprounzip/reprounzip/unpackers/common/misc.py#L94-L111